elastic/beats
# Enable monitoring buffer for elastic-agent

## What does this PR do?

Enable the monitoring buffer for the elastic-agent and beats that it starts.
Add metrics collection (from buffers) into the `diagnostics collect` bundle if enabled.

## Why is it important?

Allows us to gather metrics data for debugging if agent->ES communications are failing.

## Checklist

- [x] I have made corresponding changes to the documentation [PR](https://github.com/elastic/observability-docs/pull/1627)
- [] ~~I have added tests that prove my fix is effective or that my feature works~~

## How to test this PR locally

Start agent and run `elastic-agent diagnostics collect`. Metrics should not be included in the archive by default.
Enable metrics buffers by adding `agent.monitoring.http.buffer.enabled: true` then rerun `elastic-agent diagnostics collect`, metrics (json) files should be added to the archive

## Related issues

- Relates #29776
